client/src/dashboard/QuerySuggestionRewritingPath.js

```diff
@@ -53,7 +57,16 @@ const QuerySuggestionRewritingPath = NiceModal.create(({ queryId }) => {
 
   // call getSuggestionRewritePath() only once after initial rendering
   React.useEffect(() => {getSuggestionRewritingPath(queryId)}, []);
-  
+
+  const AddToMine = (recommendedQuery) => {
```

Comment:
This part seems a bit over-complex. Each `Add To Mine` button should be correlated to a already existent rule in the rule base (although owned by another user). Then why do we need to pass the queries to the `RewritingRuleModal`? In this way, the user has to click the `formulate` button again to generate a rule, which we already have in the database. This seems not a bit confusing. I suggest leaving this PR to its original scope that does not include the `AddToMine` button changes and merge it first. Then create a new PR for the `AddToMine` button. How does that sound?